### PR TITLE
RHBPMS-1289: Project with persistable data objects added to persistence descriptor cannot be deployed

### PR DESCRIPTION
**Pull request by wmedvede**
- Controll added in order to generate the missing identifier when required

### CODE REVIEW

**Comment by wmedvede** [comment]
@manstis 
Hello Mike would you mind review this PR. It should included for tomorrows 6.3.1 patch.

Thanks!
Walter.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java [hunk 1]
```diff
@@ -100,10 +112,47 @@ protected void loadDataObject( DataObject dataObject ) {
     public void onEntityFieldChange( String newValue ) {
         if ( getDataObject() != null ) {
 
-            Boolean doAdd = Boolean.TRUE.toString().equals( newValue );
-            DataModelCommand command = commandBuilder.buildDataObjectAddOrRemoveAnnotationCommand( getContext(),
-                    getName(), getDataObject(), JPADomainAnnotations.JAVAX_PERSISTENCE_ENTITY_ANNOTATION, doAdd );
-            command.execute();
+            final Boolean doAdd = Boolean.TRUE.toString().equals( newValue );
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java]: ```Boolean.parse(String)``` is probably nicer, but not essential.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java [hunk 2]
```diff
@@ -208,4 +270,116 @@ private void updatePropertyEditorField( String fieldId, Annotation currentValue,
...
+
+        commandBuilder.buildFieldAnnotationAddCommand( getContext(),
+                getName(),
+                dataObject,
+                idField,
+                sequenceGenerator ).execute();
+    }
+
+    private String generateIdFieldName( DataObject dataObject ) {
+        String id = "id";
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java]: This returns ```id```, ```id2```, ```id3``` is this what you really want?

**Comment by manstis** [comment]
@wmedvede I couple of minor comments.. the only troublesome one is possibly the ```id``` generator.. have a think, and fix if you agree.. otherwise +1 to merge.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java (hunk 2 above)
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java]: The idea is.
Try to generate the by default identifier with name "id". If this name already exists, then try with "id1", "id2", and so on...
99% of cases the generated id will be "id".

But the algorithm works well and generates the sequence id, id1, id2, ... etc.

Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/jpadomain/JPADataObjectEditor.java]: Great thanks for clarifying (and pointing out my interpretation was wrong anyway!!)

**Comment by manstis** [comment]
+1 to merge (when green)

**Pull request merged**